### PR TITLE
Add MockServer health check

### PR DESCRIPTION
**Pull request by woozny**

### CODE REVIEW

File: src/main/java/house/inksoftware/systemtest/domain/config/infra/InfrastructureLauncher.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public SystemTestConfiguration launchAllInfra(EmbeddedKafkaBroker kafkaBroker,
             } else if (key.equals("mockedServer")) {
                 String path = (String) ((LinkedHashMap) value).get("path");
                 Object warmupSeconds = ((LinkedHashMap) value).get("warmupSeconds");
-                launchMockedServer(path, warmupSeconds == null ? MOCKED_SERVER_DEFAULT_WARMUP_SECONDS : (int) warmupSeconds);
+                launchMockedServer(path, warmupSeconds == null ? MOCKED_SERVER_DEFAULT_WARMUP_SECONDS : (int) warmupSeconds, restTemplate);
```
Comment (IvanNikolaychuk) [inline src/main/java/house/inksoftware/systemtest/domain/config/infra/InfrastructureLauncher.java:52]: if we can reliably detect that container is up and running there's no need  for MOCKED_SERVER_DEFAULT_WARMUP_SECONDS concept

File: src/main/java/house/inksoftware/systemtest/domain/config/infra/mock/SystemTestMockedServerLauncher.java [hunk 2]
```diff
@@ -1,64 +1,126 @@
...
 
             container = new GenericContainer("mockserver/mockserver")
-                    .withCopyFileToContainer(MountableFile.forClasspathResource(relativePathToConfig), "/config/" + fileName)
+                    .withCopyFileToContainer(MountableFile.forHostPath(updatedConfigFile.getPath()), "/config/" + updatedConfigFile.getName())
                     .withEnv(envVariables);
 
             container.setPortBindings(Arrays.asList("1080:1080/tcp"));
+            container.withReuse(true);
             container.start();
             Thread.sleep(warmupSeconds * 1000);
```
Comment (IvanNikolaychuk) [inline src/main/java/house/inksoftware/systemtest/domain/config/infra/mock/SystemTestMockedServerLauncher.java:64]: why sleep if you'll waitUntilEndpointsAreAvailable?

Comment (woozny) [reply src/main/java/house/inksoftware/systemtest/domain/config/infra/mock/SystemTestMockedServerLauncher.java:64]: ah, it was for local testing only :D

File: src/main/java/house/inksoftware/systemtest/domain/config/infra/mock/SystemTestMockedServerLauncher.java [hunk 3]
```diff
@@ -1,64 +1,126 @@
...
+            waitUntilEndpointsAreAvailable(warmupSeconds);
             LOGGER.info("Starting test service mock...");
         }
     }
 
+    private void waitUntilEndpointsAreAvailable(int warmupSeconds) {
+        try {
+            await()
+                    .atMost(Duration.of(warmupSeconds, ChronoUnit.SECONDS))
```
Comment (IvanNikolaychuk) [inline src/main/java/house/inksoftware/systemtest/domain/config/infra/mock/SystemTestMockedServerLauncher.java:73]: Let's hardcode to 20 sec

**Pull request merged**